mmv1/third_party/terraform/tests/resource_bigquery_dataset_access_test.go

```diff
@@ -100,6 +100,43 @@ func TestAccBigQueryDatasetAccess_authorizedDataset(t *testing.T) {
 	})
 }
 
+func TestAccBigQueryDatasetAccess_authorizedRoutine(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"public_dataset":  fmt.Sprintf("tf_test_public_dataset_%s", randString(t, 10)),
+		"public_routine":  fmt.Sprintf("tf_test_public_routine_%s", randString(t, 10)),
+		"private_dataset": fmt.Sprintf("tf_test_private_dataset_%s", randString(t, 10)),
+	}
+
+	expected := map[string]interface{}{
+		"dataset": map[string]interface{}{
```

Comment:
Okay I think this plus the above changes should do this. Basically, the nesting inside `dataset` here and the `targetTypes` field below seem to be left over from a different test. For this test, the expected structure is:

```

	expected := map[string]interface{}{
		"routine": map[string]interface{}{
			"projectId": getTestProjectFromEnv(),
			"datasetId": context["public_dataset"],
			"routineId": context["public_routine"],
		},
	}
```